File: c_topic/src/publisher.cpp
#include<ros/ros.h>

// メッセージの型を決定(重要)
#include<std_msgs/String.h>

int main(int argc, char** argv){

    // ノードの名前を定義
    ros::init(argc,argv,"simple_talker");
    ros::NodeHandle n;

    // メッセージの型、名前、サイズを決定
    ros::Publisher ros_pub = n.advertise<std_msgs::String>("chatter",10);
    ros::Rate loop_rate(10);

    while(ros::ok()){
        std_msgs::String msg;
        msg.data = "hello World";
        
        // パブリッシュ
        ros_pub.publish(msg);

        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}

File: rospy_topic/src/publisher.py
#!/usr/bin/env python

# rospyを使うことを宣言(pythonなら必須)
import rospy
# メッセージの型を宣言(重要)
from std_msgs.msg import String

# nodeの名前を決定(なんでもよい)
rospy.init_node("publisher")

# メッセージの名前や型、サイズを決定
pub = rospy.Publisher("chatter",String,queue_size = 10)

# メッセージを送る周期を決定
r = rospy.Rate(10)

while not rospy.is_shutdown():

    # パブリッシュを実行
    pub.publish("hello")
    r.sleep()